internal/backends/mcu/fonts.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright © SixtyFPS GmbH <info@sixtyfps.io>
+// SPDX-License-Identifier: (GPL-3.0-only OR LicenseRef-SixtyFPS-commercial)
+
+use alloc::vec::Vec;
+use core::cell::RefCell;
+use core::convert::TryFrom;
+
+#[cfg(all(not(feature = "std"), feature = "unsafe_single_core"))]
+use i_slint_core::thread_local_ as thread_local;
+
+use i_slint_core::graphics::{BitmapFont, BitmapGlyph, BitmapGlyphs, FontRequest, Size};
+
+thread_local! {
+    static FONTS: RefCell<Vec<&'static BitmapFont>> = RefCell::default()
+}
+
+pub const DEFAULT_FONT_SIZE: u16 = 12;
+
+pub fn match_font(request: &FontRequest) -> (&'static BitmapFont, &'static BitmapGlyphs) {
+    let font = FONTS.with(|fonts| {
+        let fonts = fonts.borrow();
+        let fallback_font =
+            *fonts.first().expect("internal error: cannot render text without fonts");
+
+        request.family.as_ref().map_or(fallback_font, |requested_family| {
+            fonts
+                .iter()
+                .find(|bitmap_font| {
+                    core::str::from_utf8(bitmap_font.family_name.as_slice()).unwrap()
```

Comment:
I think, long term, we should not do the font lookup of familly at runtime and have the property on text already be a resolved font index or something.

But let's keep it as is for now.

Reply:
Yes, we should try to resolve at compile time what we can.